### PR TITLE
Bump @metamask/controllers version, remove patches

### PR DESCRIPTION
**Pull request by adonesky1**
Bump `@metamask/controllers` (28.0.0 -> 29.0.1) and `eth-keyring-controller` (6.2.0 -> 7.0.2) versions, remove patches that are now commits in the relevant dependencies as of these new versions.

`Bip39` patch replaced by this [commit](https://github.com/MetaMask/bip39/commit/a467f730103c08ea39d1573585d6a4421e0345ad) in our own fork of BIP39, used by `eth-hd-keyring` now

`eth-hd-keyring` patch replaced by this [commit](https://github.com/MetaMask/eth-hd-keyring/commit/e7e5ff88f9fa6531ad5209fbb9ac90be1ac6e7e8) to that package.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1954,16 +1954,14 @@ export default class MetamaskController extends EventEmitter {
    * Create a new Vault and restore an existent keyring.
    *
    * @param {string} password
-   * @param {number[]} encodedSeedPhrase - The seed phrase, encoded as an array
+   * @param {string} seedPhrase - The seed phrase, encoded as an array
    * of UTF-8 bytes.
```
Comment (digiwand) [inline app/scripts/metamask-controller.js:1958]: nit: we can update the comment description now that we've updated the type passed in

**Comment by GuillaumeRx** [comment]
> Looks like there are a few other changes from this [commit](https://github.com/MetaMask/metamask-extension/commit/98f187c301176152a7f697e62e2ba6d78b018b68#diff-e7836b5b39fa5e848534a2a3bece430e7dec9fe54b89a5bf6bcf8086975e6e26R14-R15) to revert:
>
> https://github.com/MetaMask/metamask-extension/commit/98f187c301176152a7f697e62e2ba6d78b018b68#diff-3c58b916cefff81e4927a9afd392acadbf4515f99f41783964a9851b7774791bR205-R207

This is still relevant since the controller still returns an array

**Review by GuillaumeRx — DISMISSED** [review]
LGTM !

**Review by NiranjanaBinoy — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by ryanml — APPROVED** [review]

File: patches/eth-keyring-controller+6.2.1.patch [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
-+   * @param {string|Array<number>} seedPhrase - The BIP39-compliant seed phrase,
-+   * either as a string or an array of UTF-8 bytes that represent the string.
-    * @returns {Promise<Object>} A Promise that resolves to the state.
-    */
--  createNewVaultAndRestore (password, seed) {
-+  createNewVaultAndRestore(password, seedPhrase) {
-+    const seedPhraseAsBuffer = typeof seedPhrase === 'string'
-+      ? Buffer.from(seedPhrase, 'utf8')
-+      : Buffer.from(seedPhrase)
-+
```
Comment (adonesky1) [inline patches/eth-keyring-controller+6.2.1.patch:19]: This ☝️ 🤔

**Pull request merged**